nuxeo-core/nuxeo-core-schema/src/main/java/org/nuxeo/ecm/core/schema/SchemaManagerImpl.java

```diff
@@ -1000,4 +1011,10 @@ protected String cleanPath(String path) {
         // remove index from path - we're only interested in sth/index/sth because we can't add info on sth/* property
         return path.substring(path.lastIndexOf(':') + 1).replaceAll("/-?\\d+/", "/*/");
     }
+
+    @Override
+    public List<DocumentType> getSpecialDocumentTypes() {
+        checkDirty();
+        return specialDocumentTypes.values().stream().collect(Collectors.toList());
```

Comment:
This means that `specialDocumentTypes` could be just a `List<DocumentType>`, no need for a `Map`.

Reply:
As you said in [this other comment](https://github.com/nuxeo/nuxeo/pull/3375#discussion_r327141971)
In the end it returns a set of Strings  to make fast contains predicates.
cf `DBSTransactionState#getSpecialChildrenIds`
When needed, this is joined into a list.